src/plugins/intel_npu/src/backend/src/zero_pipeline.cpp

```diff
@@ -166,23 +170,27 @@ struct IntegratedPipeline final : public Pipeline {
                        std::shared_ptr<zeroProfiling::NpuInferProfiling> npu_profiling,
                        CommandQueue& command_queue,
                        const uint32_t& group_ordinal,
-                       std::unordered_map<std::string, TensorData>& tensors_data,
-                       const size_t batch_size)
+                       const std::vector<std::optional<TensorData>>& inputTensorsData,
+                       const std::vector<std::optional<TensorData>>& outputTensorsData,
+                       const size_t numberOfCommandLists)
         : _config(config),
           _executor(static_cast<const ZeroExecutor*>(executorPtr.get())),
           _command_queue{command_queue},
-          _event_pool{device_handle, context, batch_size ? static_cast<uint32_t>(batch_size) : 1, _config},
+          _event_pool{device_handle,
+                      context,
+                      numberOfCommandLists ? static_cast<uint32_t>(numberOfCommandLists) : 1,
```

Comment:
Renamed `batch_size` to `numberOfCommandLists` inside the pipeline classes. I've found the former to be a bit confusing since there scenarios where it is lying. E.g. the real batch size could be > 1 and yet the value shown here would be 1 if the compiler is handling the batch axis. I'm thinking `numberOfCommandLists` is closer to the real description of this variable inside the pipelines.

@pereanub Lemme know if you disagree or if you have a preference.